src/test/fuzz/parse_univalue.cpp

```diff
@@ -67,7 +67,7 @@ FUZZ_TARGET(parse_univalue, .init = initialize_parse_univalue)
     } catch (const std::runtime_error&) {
     }
     try {
-        (void)ParseSighashString(univalue);
+        if (univalue.isNull() || univalue.isStr()) (void)ParseSighashString(univalue);
```

Comment:
Ooh actually, would this be better? Equally clear (I think), but we still feed it unexpected types.

```diff
diff --git a/src/test/fuzz/parse_univalue.cpp b/src/test/fuzz/parse_univalue.cpp
index a3d6ab6375..df3410eab8 100644
--- a/src/test/fuzz/parse_univalue.cpp
+++ b/src/test/fuzz/parse_univalue.cpp
@@ -67,8 +67,10 @@ FUZZ_TARGET(parse_univalue, .init = initialize_parse_univalue)
     } catch (const std::runtime_error&) {
     }
     try {
-        if (univalue.isNull() || univalue.isStr()) (void)ParseSighashString(univalue);
+        (void)ParseSighashString(univalue);
     } catch (const UniValue&) {
+    } catch (std::runtime_error&) {
+        if(univalue.isStr()) throw;  //a non-null, non-string univalue is expected to throw on .get_str()
     }
     try {
         (void)AmountFromValue(univalue);

```

Reply:
Might also want to catch it NOT throwing an error when it ought to.